src/java/boa/compiler/BoaCompiler.java

```diff
@@ -400,6 +411,7 @@ private static CommandLine processCommandLineOptions(final String[] args) {
 		options.addOption("n", "name", true, "the name of the generated main class");
 		options.addOption("ast", "ast-debug", false, "print the AST after parsing and before code generation (debug)");
 		options.addOption("pp", "pretty-print", false, "pretty print the AST before code generation (debug)");
+		options.addOption("cd", "compilation-dir", true, "All generated Files live here");
```

Comment:
I think this could be a general/global option.  Right now we generate into /tmp (or whatever temp dir your OS uses) and then make unique sub-dirs there.  That could be the default, and then this option would allow pointing somewhere specific.